compiler/rustc_ast/src/token.rs

```diff
@@ -229,35 +229,61 @@ fn ident_can_begin_type(name: Symbol, span: Span, is_raw: bool) -> bool {
 #[derive(PartialEq, Encodable, Decodable, Debug, HashStable_Generic)]
 pub enum TokenKind {
     /* Expression-operator symbols. */
+    /// `=`
```

Comment:
I have often wondered about the meaning of some variants, so let's just comment what they actually are.

Reply:
This change and the `// tidy-alphabetical` stuff can be submitted separately and I'll approve them right away.